diff --git a/bottle.py.orig b/bottle.py
index 54ee51d..e88bdb7 100644
--- a/bottle.py.orig
+++ b/bottle.py
@@ -2459,6 +2459,11 @@ def _file_iter_range(fp, offset, bytes, maxread=1024*1024):
         if not part: break
         bytes -= len(part)
         yield part
+    try:
+        fp.close()
+    except:
+        import traceback
+        traceback.print_exc(file=sys.stderr)
 
 
 def static_file(filename, root, mimetype='auto', download=False, charset='UTF-8'):
